jetson/nav/filter.hpp

```diff
@@ -0,0 +1,72 @@
+#pragma once
+
+#include <array>
+#include <numeric>
+#include <algorithm>
+
+template<typename T>
+class Filter {
```

Comment:
Just for my sake, this is to replace the caching we were doing before, or is this for a diff reason?